src/Tools/ExternalAccess/LiveShare/Tagger/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.Implementation.Classification;
+using Microsoft.CodeAnalysis.Editor.Shared.Threading;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ExternalAccess.LiveShare.Classification;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LiveShare.Tagger
+{
+    /// <summary>
+    /// this is almost straight copy from typescript for syntatic LSP experiement.
+    /// we won't attemp to change code to follow Roslyn style until we have result of the experiement
+    /// </summary>
+    internal sealed partial class SyntacticClassificationTaggerProvider
+    {
+        /// <summary>
+        /// Backing implementation for all <see cref="Tagger"/> instances for a single <see cref="ITextBuffer"/>.
+        /// </summary>
+        internal partial class TagComputer
```

Comment:
seems strange that we're reeimplementing hte tag computer.  why can't this use the existing async tagging infrastructure, and only provide the abstract tagger impl?

Reply:
I think you didn't read comment at all on code. we are not interested in clean up these code from typescript now, we are interested in doing an experiment based on this approach. if this approach is proved to be one we will take. then the platform team implements actual tagger.